src/connection.rs

```diff
@@ -559,24 +561,20 @@ impl<T: AsyncRead + AsyncWrite + Unpin> Connection<T> {
         Ok(())
     }
 
-    async fn send(&mut self, f: impl Into<Frame<()>>) -> Result<()> {
-        send(self.id, &mut self.socket, &mut self.stream_receiver, &mut self.control_receiver, f).await
-    }
-
     /// Process a command from one of our `Stream`s.
     async fn on_stream_command(&mut self, cmd: Option<StreamCommand>) -> Result<()> {
         match cmd {
             Some(StreamCommand::SendFrame(frame)) => {
                 log::trace!("{}/{}: sending: {}", self.id, frame.header().stream_id(), frame.header());
-                self.send(frame).await.or(Err(ConnectionError::Closed))?
+                self.socket.get_mut().feed(frame.into()).await.or(Err(ConnectionError::Closed))?
```

Comment:
```suggestion
                self.socket.feed(frame.into()).await.or(Err(ConnectionError::Closed))?
```
I don't think we need `.get_mut()` as we're using `Stream` APIs.